### ISSUE TITLE
Nested f.inputs blocks

### ISSUE THREAD

**Issue by HoneyryderChuck** [justinfrench/formtastic#848]
Hi,

I'm having a different behaviour in my project updating from the formtastic for rails 2 version (1.x) to the one for rails 3 (2.2.0) concerning nested f.input blocks. We previously had something like this:

<pre>
<code>
- f.inputs do
%lt;li class="bang"%gt;
 - f.inputs do
= f.input :bla, .....
</code>
</pre>

that would nest ol elements. I don't recall if the previous version nested ols inside ols without setting an in-between li (maybe that was the reason for the update, and why somebody wrote that li in between). 

<pre>
<code>
%lt;ol%gt;
%lt; li class="bang%gt;
%lt; ol%gt;
%lt;li class="input%gt;
%lt;input ..../%gt;
%lt;/li%gt;
%lt;/ol%gt;
%lt;/li%gt; 
%lt;/ol%gt;
</code>
</pre>

All i know is that, if I do the same with the newest version, the ols don't get nested anymore, but an empty li is created instead, before the input element gets written. 

<pre>
<code>

%lt;ol%gt;
%lt;li class="bang"%gt;
%lt;li class="input%gt;
%lt;input .../%gt;
%lt;/li%gt;
%lt;/ol%gt;

</code>
</pre>

Well, I don't have anything against it not nesting the list using the same f element (I do think it is cleaner this way), but I end up with an empty li I don't have an use for anymore and an "ignored" second f.inputs call. 

Basically, in such cases, I would rather have a warning message in the log, or an explicit Exception thrown. This would be especially important for projects updating to rails 3, whose forms will silently "fail" in such cases, the styling remaining as the only way to actually perceive the different behaviour. 

Unless there is a reason why this was made this way that I'm not following.

Regards,
Tiago
